apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/amqp_data_consumer.ex

```diff
@@ -94,64 +101,85 @@ defmodule Astarte.DataUpdaterPlant.AMQPDataConsumer do
 
   # Server callbacks
 
+  @impl true
   def init(args) do
     queue_name = Keyword.fetch!(args, :queue_name)
-    # Init asynchronously since we will wait for the AMQP connection
-    send(self(), {:async_init, queue_name})
-    {:ok, nil}
+    {:ok, %State{queue_name: queue_name}, {:continue, :connect}}
```

Comment:
Does the async initialization provide any actual value or is this something we could simply remove while we're at it?

Reply:
The AMQPDataConsumer may handle some `call`s while it is not consuming from a queue (e.g. [`start_messag_tracker`](https://github.com/astarte-platform/astarte/blob/7ff7557ceb4d511e6761e9d828ebb641f942f4ba/apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/amqp_data_consumer.ex#L69-L73)), so the async init provides a way to do that even if it is still setting up the consuming  part